samples/scala-replicated-examples/src/main/scala/com/example/replicated/counter/CounterHandler.scala

```diff
@@ -0,0 +1,27 @@
+package com.example.replicated.counter
+
+import com.akkaserverless.replicatedentity.ReplicatedData
+import com.akkaserverless.scalasdk.impl.replicatedentity.ReplicatedEntityHandler
+import com.akkaserverless.scalasdk.replicatedentity.CommandContext
+import com.akkaserverless.scalasdk.replicatedentity.ReplicatedCounter
+import com.akkaserverless.scalasdk.replicatedentity.ReplicatedEntity
+
+class CounterHandler(entity: Counter) extends ReplicatedEntityHandler[ReplicatedCounter, Counter](entity) {
+
+  override def handleCommand(
+      commandName: String,
+      data: ReplicatedData,
+      command: Any,
+      context: CommandContext): ReplicatedEntity.Effect[_] = {
+
+    val scalaData = data.asInstanceOf[ReplicatedCounter]
```

Comment:
The Scala `handleCommand` can't receive a `D <: ReplicatedData` as the java counterpart, it needs to be typed to `ReplicatedData`. 

The reason is that we the SDK works with Java types. So, for example, it will pass a Java ReplicatedCounter and we need to wrap it as in `ScalaReplicatedCounter(JavaReplicatedCounter)` but if we type it to `D`, then D is `JavaReplicatedCounter`, not `ScalaReplicatedCounter`. 

Therefore, we need to type it to `ReplicatedData` and cast it to the right type inside the handle. I'm not super happy with this solution and want to massage that part of the code, but I doubt we can do much.